injector.js

```diff
@@ -142,8 +151,8 @@ let storeData = function(repository, chain, address, compilationResult, sources)
   } else if (cborData['bzzr1']) {
     metadataPath = '/swarm/bzzr1/' + Web3.utils.bytesToHex(cborData['bzzr1']).slice(2)
   } else if (cborData['ipfs']) {
-    throw "TODO: Implement ipfs"
-    //metadataPath = '/ipfs/' + Web3.utils.bytesToHex(cborData['ipfs']).slice(2)
+    const bytecodeLink = new dagPB.DAGLink('', 0, cborData['ipfs']);
```

Comment:
What does this result in?

Reply:
The link is an object like this:
```
DAGLink {
  _name: '',
  _nameBuf: null,
  _size: 0,
  _cid:
   CID {
     version: 0,
     codec: 'dag-pb',
     multihash:
      <Buffer 12 20 27 dd 4c f1 4e 24 03 b8 43 ff 81 26 5c ce a9 0b d3 be 07 ef c0 d0 bb 7c 2d 8c e2 76 fa c7 df 58>,
     multibaseName: 'base58btc' } 
}
```

It looks like the conversion could be done more simply (without creating a link) with: 
```javascript
multihashes.toB58String(cborData['ipfs']);

> QmR2GL3J5ggeEKusoFgtkRNzKd5BXJt43nYQoWN9Ay9hXD
```

**Libs**
+ [multihashes](https://github.com/multiformats/js-multihash#js-multihash)
+ [ipld-dab-pb DAGLink docs](https://github.com/ipld/js-ipld-dag-pb#daglink-functions).

Reply:
Simplified to use only the multihashes form in 754add8a7569cf7bcfdf1ec4d8600a63f51cac03